crypto/include/internal/rand_int.h

```diff
@@ -58,7 +58,8 @@ void rand_crngt_cleanup_entropy(RAND_DRBG *drbg,
 /*
  * RAND_POOL functions
  */
-RAND_POOL *rand_pool_new(int entropy_requested, size_t min_len, size_t max_len);
+RAND_POOL *rand_pool_new(int entropy_requested, int secure,
+                         size_t min_len, size_t max_len);
```

Comment:
It just occurred to me that we don't need a separate `secure` parameter, because there is a simple rule: Allocation from secure memory is required if and only if `entropy_requested > 0`. I checked the source, this rule holds for all entropy, nonce, and additional data callbacks which use the pool. You can easily verify it using the `-W (--function-context)` option of `git grep`:


```
git grep -W rand_pool_new
```

The only exception of this rule is - of all things - the FIPS CRNGT, which was added in d69226a3fc8e:


https://github.com/openssl/openssl/blob/efa4add59b6bf6ff2c84d7710404cfbadeb353c7/crypto/rand/rand_crng_test.c#L48


But I think that this exception from the rule can be fixed in a simple way by passing 1 (in fact, any number between 1 and the blocksize 128) as `entropy_requested` parameter. @paulidale what do you think about my suggestion?

Reply:
More precisely, any number between 1 and 8 * CRNGT_BUFSIZ.

Reply:
This seems like a good approach.  It is the entropy that is being protected.

Reply:
Yes, but, RAND_DRBG_new creates non-secure memory DRBG, which
will then be in non-secure memory but only the entropy buffer is in secure memory.
That felt a bit like broken symmetry to me.
Of course _we_ do not use RAND_DRBG_new.

Reply:
> Yes, but, RAND_DRBG_new creates non-secure memory DRBG

Hmmm... if you are right (I didn't look) then that should be checked. At least the internal state of the DRBG should be placed into secure memory, otherwise it makes no sense to do it for the entropy input either.

Reply:
> That felt a bit like broken symmetry to me.

I agree that it should be consistent.